### PR TITLE
[BE] Get list of job applicants

### PR DESCRIPTION
**Pull request by nspc97**
[BE] Get list of job applicants
https://trello.com/c/ycA4ZSKm/51-be-get-list-of-job-applicants

### CODE REVIEW

**Review by MihaiDruta — APPROVED** [review]

File: src/Controller/ActivityController.php [hunk 1]
```diff
@@ -479,4 +486,25 @@ public function applyForActivity(Activity $activity, ActivityUserRepository $act
         $activityUserRepo->apply($activity, $applierUser);
         return new JsonResponse(['message' => 'Applied with success!'], Response::HTTP_OK);
     }
+
+    /**
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php:484]: Missing `@SWG\Schema`

File: src/Controller/ActivityController.php [hunk 2]
```diff
@@ -479,4 +486,25 @@ public function applyForActivity(Activity $activity, ActivityUserRepository $act
...
+     *Apply for an Activity.
+     * @Rest\Get("/{id}/applicants")
+     * @param Activity $activity
+     * @param ActivityUserRepository $activityUserRepo
+     * @return JsonResponse
+     */
+    public function listOfApplicants(Activity $activity, ActivityUserRepository $activityUserRepo): JsonResponse
+    {
+        $applicants = $activityUserRepo->getApplicantsForActivity($activity);
+        $context = SerializationContext::create();
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: You are sure you don't need a context? What info is returned about the applicants?

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Review by MihaiDruta — APPROVED** [review]

File: src/Controller/ActivityController.php [hunk 3]
```diff
@@ -488,16 +482,49 @@ public function applyForActivity(Activity $activity, ActivityUserRepository $act
...
+     *     required=true,
+     *     type="integer",
+     * )
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Successfull operation!",
+     *     @SWG\Schema(
+     *     type="array",
+     *     @Model(type=User::class, groups={"UserDetail"}),
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: `UserList` should be used as there's no need to show all the details about the user. Make sure it exists.

File: src/Controller/ActivityController.php [hunk 4]
```diff
@@ -488,16 +482,49 @@ public function applyForActivity(Activity $activity, ActivityUserRepository $act
...
      * @param ActivityUserRepository $activityUserRepo
      * @return JsonResponse
      */
     public function listOfApplicants(Activity $activity, ActivityUserRepository $activityUserRepo): JsonResponse
     {
         $applicants = $activityUserRepo->getApplicantsForActivity($activity);
-        $context = SerializationContext::create();
+
+        /** @var SerializationContext $context */
+        $context = SerializationContext::create()->setGroups(array('UserDetail'));
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: UserDetail -> UserList

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Comment by SpartakusMd** [comment]
Add the alias UserList in https://github.com/p5chi-auf/PentaBee-api/blob/develop/config/packages/nelmio_api_doc.yaml

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**